src/plugins/intel_gna/src/optimizer/gna_pass_manager.cpp

```diff
@@ -2582,6 +2587,116 @@ void TransposeWeightsFromNCHWToNHWCPass::run() {
     }
 }
 
+void FuseFullyConnectedWithEltwisePass::run() {
+    // This legacy pass removes the Eltwise (only if it performs SUM op) from between FC and Any.
+    // The Const layer blob data is added to biases blob data of FC layer.
+    // Finally Const and Permute are also removed
+    // Reshape (if exist) is moved after Any to keep output shape unchanged.
+    //
+    // This operation can be illustrated as follows:
+    //
+    // --
+    //  Original:                  Result:           Removed:
+    //
+    //    FC                         FC              (Eltwise)
+    //    |                          |               (Const)
+    //  Permute                   Reshape            (Permute)
+    // (optional)           (if exists in Original)
+    //    |                          |
+    //  Reshape                     Any
+    // (optional)  Const
+    //    |       /
+    //  Eltwise(sum)
+    //    |
+    //  Any (e.g. ReLU)
+    //--
+    //
+    // NOTE: This pass is implemented to prevent unnecessary roundtrip to memory (additional layer).
+    // It can be fully removed if corresponding LPT transformation is already implemented.
+    OV_ITT_SCOPED_TASK(itt::domains::GNA_LT, "FuseFullyConnectedWithEltwisePass");
+    auto isEltwiseWithSum = [](CNNLayerPtr layer) {
+        return layer != nullptr && LayerInfo(layer).isEltwise() && layer->outData.size() &&
+               layer->params["operation"] == "sum";
+    };
+    for (auto& layer : *pLayers) {
+        if (!LayerInfo(layer).isFullyConnected() || !layer->outData.size()) {
+            continue;
+        }
+        CNNLayerPtr next = nullptr;
+
+        auto& fully_connected = layer;
+
+        next = CNNNetGetNextLayerSkipCertain(fully_connected, 0, 0, DoNotSkip, true).first;
```

Comment:
Why don't check inside skip function that layer is permute/reshape and IsOneDimLayer?

Reply:
agree, if it is possible to add verificiation in "skip" function instead of adding new argument checkonly, which introduce inconsistence ( in such case DoNotSkip can skip )

Reply:
Yes, it was possible. Changed.